ariadne/enums.py

```diff
@@ -49,3 +107,132 @@ def set_default_enum_values(graphql_type: GraphQLEnumType):
+        for name, input_ in schema.type_map.items()
+        if isinstance(input_, GraphQLInputObjectType)
+    )
+    for name, type_ in object_types:
+        yield from enum_values_in_type_fields(name, type_)
+
+    for name, input_ in input_types:
+        yield from enum_values_in_input_fields(name, input_)
+
+
+def enum_values_in_type_fields(
+    field_name: str,
+    type_: GraphQLObjectType,
+) -> Generator[ArgumentWithKeys, None, None]:
+    for field in type_.fields.values():
+        yield from enum_values_in_field_args(field_name, field)
+
+
+def enum_values_in_input_fields(
+    field_name,
+    input_: GraphQLInputObjectType,
+) -> Generator[InputFieldWithKeys, None, None]:
+    for input_name, field in input_.fields.items():
+        type_ = resolve_null_type(field.type)
+        if isinstance(type_, GraphQLEnumType):
+            yield field_name, input_name, field, None
+
+        if isinstance(type_, GraphQLInputObjectType):
+            if field.ast_node is not None:
+                routes = get_enum_keys_from_ast(field.ast_node)
+                for route in routes:
+                    yield field_name, input_name, field, route
+
+
+def enum_values_in_field_args(
+    field_name: str,
+    field: GraphQLField,
+) -> Generator[ArgumentWithKeys, None, None]:
+    args = [
+        (name, arg)
+        for name, arg in field.args.items()
+        if isinstance(
+            arg.type, (GraphQLInputObjectType, GraphQLEnumType, GraphQLNonNull)
+        )
+    ]
+
+    for arg_name, arg in args:
+        type_ = resolve_null_type(arg.type)
+        if isinstance(type_, GraphQLEnumType):
+            yield field_name, arg_name, arg, None
+
+        if isinstance(type_, GraphQLInputObjectType):
+            if arg.ast_node is not None and arg.ast_node.default_value is not None:
+                routes = get_enum_keys_from_ast(arg.ast_node)
+                for route in routes:
+                    yield field_name, arg_name, arg, route
+
+
+def get_enum_keys_from_ast(ast_node: InputValueDefinitionNode) -> List[List["str"]]:
+    routes = []
+    object_node = ast_node.default_value
+    object_node = cast(ObjectValueNode, object_node)
+    nodes = [([field.name.value], field) for field in object_node.fields]
+
+    while nodes:
+        key_list, field = nodes.pop()
+        if isinstance(field.value, EnumValueNode):
+            routes.append(key_list)
+
+        if isinstance(field.value, ObjectValueNode):
+            for new_field in field.value.fields:
+                new_route = key_list[:]
+                new_route.append(new_field.name.value)
+                nodes.append((new_route, new_field))
+
+    return routes
+
+
+def is_invalid_enum_value(field: Union[GraphQLInputField, GraphQLArgument]) -> bool:
+    if field.ast_node is None:
+        return False
+    return field.default_value is Undefined and field.ast_node.default_value is not None
+
+
+def validate_schema_enum_values(schema: GraphQLSchema) -> None:
```

Comment:
Move this function before `find_enum_values_in_schema`.

Reply:
Done.